core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -1897,7 +1897,7 @@ private Builder builder(RelNode rel, Set<Clause> clauses) {
       if (!selectList.equals(SqlNodeList.SINGLETON_STAR)) {
         final boolean aliasRef = expectedClauses.contains(Clause.HAVING)
             && dialect.getConformance().isHavingAlias();
-        newContext = new Context(dialect, selectList.size()) {
+        newContext = new BaseContext(dialect, selectList.size()) {
```

Comment:
how is this even possible, BaseContext is abstract?

Reply:
`Context` is also an abstract class. what is happening here is that the previous author decided to instantiate a new instance of `Context` by implementing the abstract methods on the spot which is the syntax you are seeing here: `new Context(dialect, selectList.size()) {}` (note the curly braces)

Reply:
I think this class is big enough to be pulled outside in a static named class.
(The original author should have done that, and documented what this class is supposed to do)

Reply:
sure, I can extract it. would you by any chance have a suggestion for a class name?

Reply:
I extracted the anonymous class into a nested class. Let me know if you are good with the name or if you have a better suggestion.